workers/generic-worker/livelog/livelog.go

```diff
@@ -92,6 +95,13 @@ func New(liveLogExecutable string, putPort, getPort uint16) (*LiveLog, error) {
 
 	select {
 	case err := <-inputStreamConnectionResult:
+		go func() {
+			select {
+			case <-l.done:
+			case pr := <-putResult:
+				log.Printf("WARNING: Livelog failure: error '%v' and livelog output:\n%s", pr.e, pr.b)
+			}
+		}()
```

Comment:
Launching this error-reporting in a goroutine seems weird.  Is the idea to return immediately, but as soon as the livelog process exits, display its output, even if that's later?

Reply:
The constructor needs to return immediately, so the go routine is launched to await either the closure of the `done` channel (which signifies that the logging has completed and `livelog` is about to be terminated) or for the `livelog` process to complete. If the `livelog` process completes before the `done` channel is closed, it means the livelog exited before we intentionally killed it. The output and the error returned by the `livelog` command are written to the `putResult` channel (line 83), so this new go routine's purpose is to capture the output and error of `livelog` if it completes before the `done` channel is closed, and report them to the standard error of the worker process.

This is how I discovered that the reason `livelog` was terminating early was due to the hardcoded port you added, which is fixed above in `workers/generic-worker/livelog.go`.